game/Player.cpp

```diff
@@ -7752,6 +7756,15 @@ void idPlayer::RouteGuiMouse( idUserInterface *gui ) {
 		oldMouseX = usercmd.mx;
 		oldMouseY = usercmd.my;
 	}
+	if ( usercmd.jx != 0 || usercmd.jy != 0 ) {
+		extern idCVar in_padMouseSpeed;
+		float speed = in_padMouseSpeed.GetFloat() / 640.f * renderSystem->GetScreenWidth();
+		// hack: we want the original axis deflection, without inversed axes if enabled
+		int x, y;
+		Sys_GetCombinedAxisDeflection( x, y );
```

Comment:
Question: why you added jx/jy members to usercmd_t instead of converting this data into mouse movements and writing it to mx/my?

Reply:
Because `mx` and `my` are not movements, but absolute mouse positions and therefore need to be handled quite a bit differently from joystick input. I originally wanted to do what you suggest, but it's simply not possible. At least not without some refactoring, and that seemed risky...

Reply:
This is a bit sad.
Thankfully there are not too much code depending on these coordinates, so why not =)

Reply:
Yeah... I've come to the conclusion that the UI code is really not a highlight of the engine. It's some of the less appealing code I've seen so far. A lot of it seems hacked in and not easily extensible. I think I can understand why they replaced it for the BFG version...